tests/src/Unit/ElasticSearch/Parameters/Factory/FilterFactoryTest.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Drupal\Tests\elasticsearch_connector\Unit\ElasticSearch\Parameters\Factory;
+
+use Drupal\search_api\Query\Condition;
+use Drupal\Tests\UnitTestCase;
+use Drupal\elasticsearch_connector\ElasticSearch\Parameters\Factory\FilterFactory;
+
+/**
+ * This test is long because it tests a long method. It just repeats
+ * the same pattern over and over where a condition is mocked to
+ * test each of the ramifications.
+ *
+ * @coversDefaultClass \Drupal\elasticsearch_connector\ElasticSearch\Parameters\Factory\FilterFactory
+ *
+ * @group elasticsearch_connector
+ */
+class FilterFactoryTest extends UnitTestCase {
+
+  /**
+   * @covers ::filterFromCondition
+   */
+  public function testFilterFromConditionA() {
+    // Empty and not empty operators.
+    /** @var \Prophecy\Prophecy\ObjectProphecy $condition */
+    $condition = $this->prophesize(Condition::class);
+
+    $condition->getValue()
+      ->willReturn(FALSE);
+
+    $condition->getOperator()
+      ->willReturn('<>');
+
+    $condition->getField()
+      ->willReturn('foo');
+
+    $filter = FilterFactory::filterFromCondition($condition->reveal());
+    $expected_filter = [
+      'exists' =>
+        [
+          'field' => 'foo',
+        ],
+    ];
+    $this->assertEquals($expected_filter, $filter);
+
+    // Thest the = operator.
+    /** @var \Prophecy\Prophecy\ObjectProphecy $condition */
+    $condition = $this->prophesize(Condition::class);
+
+    $condition->getValue()
+      ->willReturn(FALSE);
+
+    $condition->getOperator()
+      ->willReturn('=');
+
+    $condition->getField()
+      ->willReturn('foo');
+
+    $filter = FilterFactory::filterFromCondition($condition->reveal());
+    $expected_filter = [
+      'bool' => [
+        'must_not' => [
+          'exists' => ['field' => 'foo'],
+        ],
+      ],
+    ];
+    $this->assertEquals($expected_filter, $filter);
+
+    // Other operators will throw an exception.
+    /** @var \Prophecy\Prophecy\ObjectProphecy $condition */
+    $condition = $this->prophesize(Condition::class);
+
+    $condition->getValue()
+      ->willReturn(FALSE);
+
+    $condition->getOperator()
+      ->willReturn('>');
+
+    $condition->getField()
+      ->willReturn('foo');
+
+    $this->setExpectedException(\Exception::class, 'Incorrect filter criteria');
+    FilterFactory::filterFromCondition($condition->reveal());
+  }
+
+  /**
+   * @covers ::filterFromCondition
+   */
+  public function testFilterFromConditionB() {
+    // Normal filters
+    /** @var \Prophecy\Prophecy\ObjectProphecy $condition */
+    $condition = $this->prophesize(Condition::class);
```

Comment:
The pattern of tests here is fine, but it seems like a good fit for a `@dataProvider`, where for each test case it would pass the value, operator, field name and expected result.

Reply:
Yup, I thought that it was clear this way but there is a lot of code duplicity. I will change it into a data provider.